### PR TITLE
Add a button handler

### PR DESCRIPTION
**Pull request by Szedann**

### CODE REVIEW

File: src/handlers/button.handler.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    ButtonBuilder,
+    ButtonInteraction,
+    Events,
+    InteractionButtonComponentData,
+} from 'discord.js';
+import { Handler } from '..';
+import { BaseSchema, Output } from 'valibot';
+import { error } from 'console';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IThundxr) [inline src/handlers/button.handler.ts:11]: I don't think we need to use any here

File: src/handlers/button.handler.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const buttonMap = new Map<string, Button<any>>();
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IThundxr) [inline src/handlers/button.handler.ts:14]: Here aswell

**Review by IThundxr — COMMENTED** [review]
I'll further look at this from PC

File: src/handlers/button.handler.ts (hunk 2 above)
Comment (Szedann) [reply src/handlers/button.handler.ts:14]: I actually think we do

Comment (Szedann) [reply src/handlers/button.handler.ts:14]: but I'm gonna look into it a bit more

Comment (Szedann) [reply src/handlers/button.handler.ts:14]: tried to "fix". Those `any`s are for the greater good

Comment (Szedann) [reply src/handlers/button.handler.ts:14]: btw it is typesafe still, the class enforces a certain generic and valibot verifies the id

**Pull request merged**